### PR TITLE
Readme: Include module dependencies in table

### PR DESCRIPTION
**Pull request by Mottie**
Include a list of module dependencies in the main readme. Also add note in the header that modules need to be loaded in order.

Ref gh-543

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,16 +142,18 @@ information on its usage.
...
-| globalize/date.js | 5.1KB | 3.8KB | [Date module](#date-module) provides date formatting and parsing |
-| globalize/message.js | 5.4KB | 0.7KB | [Message module](#message-module) provides ICU message format support |
-| globalize/number.js | 3.1KB | 1.8KB | [Number module](#number-module) provides number formatting and parsing |
-| globalize/plural.js | 2.3KB | 0.4KB | [Plural module](#plural-module) provides pluralization support |
-| globalize/relative-time.js | 0.8KB | 0.6KB | [Relative time module](#relative-time-module) provides relative time formatting support |
-| globalize/unit.js | 0.9KB | 0.5KB | [Unit module](#unit-module) provides unit formatting support |
+| File | Minified + gzipped size | Runtime minified + gzipped size | Dependencies (load in order) | Summary |
+|---|--:|--:|---|---|
+| globalize.js | 1.5KB | 1.0KB | Cldrjs & CLDR-data | [Core library](#core-module) |
+| globalize/currency.js | 2.6KB | 0.6KB | core, number.js & plural.js | [Currency module](#currency-module) provides currency formatting and parsing |
```
Comment (jzaefferer) [inline README.md]: "core" is inconsistent with the other references (which have the ".js" extension). Not sure if "globalize.js" is much better though.

**Comment by jzaefferer** [comment]
Overall looks pretty good. I feel like there's room for improvement, but not sure how. Hopefully @rxaviers has some ideas.

**Comment by rxaviers** [comment]
Sorry my review took so long. I liked your change, but I would rather not change this table. Instead, I would place this information in the dependencies section. Can you revert the change from this summary table and create a new table under dependencies?

What do you think of this change?

**Comment by Mottie** [comment]
That sounds good to me, as long as the information is there I'll be happy :smile:

**Comment by rxaviers** [comment]
Thanks!

File: README.md [hunk 2]
```diff
@@ -153,6 +153,8 @@ information on its usage.
 | globalize/relative-time.js | 0.8KB | 0.6KB | [Relative time module](#relative-time-module) provides relative time formatting support |
 | globalize/unit.js | 0.9KB | 0.5KB | [Unit module](#unit-module) provides unit formatting support |
 
+Use [this page](http://johnnyreilly.github.io/globalize-so-what-cha-want/) for more details on which Cldrjs file & CLDR data to include.
+
```
Comment (rxaviers) [inline README.md]: Can you move this sentence down too? Please, also consider using words parallel with https://github.com/jquery/globalize/pull/547/files#diff-04c6e90faac2675aa89e2176d2eec7d8R235.

**Comment by Mottie** [comment]
Ok, how does this look to you?

| Module | Dependencies (load in order) |
|---|---|
| Core module | Cldrjs & CLDR data |
| Currency module | globalize.js, globalize/number.js, globalize/plural.js & globalize/currency.js |
| Date module | globalize.js, globalize/number.js & globalize/date.js |
| Message module | globalize.js & globalize/message.js |
| Number module | globalize.js & globalize/number.js |
| Plural | globalize.js & globalize/plural.js |
| Relative time module | globalize.js, globalize/number.js, globalize/plural.js & globalize/relative-time.js |
| Unit module | globalize.js, globalize/number.js, globalize/plural.js & globalize/unit.js |

Do you think it should include the module file at the end?

**Comment by rxaviers** [comment]
> | Module | Dependencies (load in order) |
> |---|---|
> | Core module | Cldrjs & CLDR data |

Let's remove "CLDR data", because there's an extensive section right below explaining that dependency requirements:

```diff
- | Core module | Cldrjs & CLDR data |
+ | Core module | Cldrjs |
```

> | Module | Dependencies (load in order) |
> |---|---|
> | Currency module | globalize.js, globalize/number.js, globalize/plural.js & globalize/currency.js |
> | Date module | globalize.js, globalize/number.js & globalize/date.js |
> | Message module | globalize.js & globalize/message.js |
> | Number module | globalize.js & globalize/number.js |
> | Plural | globalize.js & globalize/plural.js |
> | Relative time module | globalize.js, globalize/number.js, globalize/plural.js & globalize/relative-time.js |
> | Unit module | globalize.js, globalize/number.js, globalize/plural.js & globalize/unit.js |

Let's replace all `&` for the verbose `, and`.

> Do you think it should include the module file at the end?

I think it doesn't need to cite the own module, because it's not a dependency, but the module itself. For example:

```diff
- | Unit module | globalize.js, globalize/number.js, globalize/plural.js & globalize/unit.js |
+ | Unit module | globalize.js, globalize/number.js, and globalize/plural.js |
```

**Comment by rxaviers** [comment]
On currency module, `globalize/plural.js` is only required for "code" or "name" styles. It's not a requirement for "symbol" or "accounting" styles.

**Comment by rxaviers** [comment]
I would also consider this one more change:
```diff
- globalize.js
+ globalize.js (core)
```

**Comment by rxaviers** [comment]
On Message module, `globalize/plural.js` is a requirement if using messages that need pluralization support.

**Comment by Mottie** [comment]
Ok, this is the result

| Module | Dependencies (load in order) |
|---|---|
| Core module | Cldrjs |
| Currency module | globalize.js (core), globalize/number.js, and globalize/plural.js (only required for "code" or "name" styles) |
| Date module | globalize.js (core) and globalize/number.js |
| Message module | globalize.js (core) and globalize/plural.js (if using messages that need pluralization support) |
| Number module | globalize.js (core) |
| Plural | globalize.js (core) |
| Relative time module | globalize.js (core), globalize/number.js, and globalize/plural.js |
| Unit module | globalize.js (core), globalize/number.js, and globalize/plural.js |

**Comment by rxaviers** [comment]
It looks great. Thanks!

**Comment by Mottie** [comment]
@johnnyreilly I just wanted to let you know about these differences, in case you want to update your site.

> On currency module, globalize/plural.js is only required for "code" or "name" styles. It's not a requirement for "symbol" or "accounting" styles.

and

> On Message module, globalize/plural.js is a requirement if using messages that need pluralization support.

**Comment by rxaviers** [comment]
By the way, I have reworded the previous paragraph of this section at 1c90d951691d4c81ea0b923c347c31d022c77556.

**Comment by johnnyreilly** [comment]
Hi @Mottie,

Thanks for highlighting; I'll try and take a look.  Put a reminder in place: https://github.com/johnnyreilly/globalize-so-what-cha-want/issues/5

**Pull request closed without merging**